stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java

```diff
@@ -122,6 +129,34 @@ public void process(K key, V1 value) {
             }
         }
 
+        private boolean joinable() {
+            Set<TopicPartition> partitions = ((ProcessorContextImpl) this.context).task().partitions();
```

Comment:
I feel this function is specific to KStream not in Processor, as its checking logic is sort-of specific to KStream's definitions. Which tests need MockProcessorContext.joinable() calls?